src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/TableSaw.java

```diff
@@ -54,28 +63,39 @@ public TableSaw(Category category, SlimefunItemStack item) {
     }
 
     @Override
-    public void onInteract(Player p, Block b) {
-        ItemStack log = p.getInventory().getItemInMainHand();
+    public void onInteract(@Nonnull Player p, @Nonnull Block b) {
+        ItemStack item = p.getInventory().getItemInMainHand();
 
-        Optional<Material> planks = MaterialConverter.getPlanksFromLog(log.getType());
+        boolean itemIsAPlank = Tag.PLANKS.isTagged(item.getType());
+        boolean itemIsALog = Tag.LOGS.isTagged(item.getType());
 
-        if (planks.isPresent()) {
-            if (p.getGameMode() != GameMode.CREATIVE) {
-                ItemUtils.consumeItem(log, true);
-            }
+        ItemStack output = null;
 
-            ItemStack output = new ItemStack(planks.get(), 8);
-            Inventory outputChest = findOutputChest(b, output);
+        if (itemIsALog) {
+            Optional<Material> planks = MaterialConverter.getPlanksFromLog(item.getType());
+            output = new ItemStack(planks.get(), 8);
+        }
+        else if (itemIsAPlank){
+            output = new ItemStack(Material.STICK, 4);
+        }
+        else {
+            return;
+        }
```

Comment:
I think this should be put into a seperate method returning a nullable ItemStack.